_posts/2017-02-02-using-discover-agrc-utah-gov-services-in-the-export-web-map-print-service.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+status: publish
+published: true
+title: 'Using discover.agrc.utah.gov Services PrintingTools/Export Web Map Service'
+author:
+  display_name: Scott Davis
+  email: stdavis@utah.gov
+date: 2017-02-02 21:04:30
+categories:
+
+tags:
+- arcgis server
+- printing
+- wmts
+
+---
+### Export Web Map Tool
+The [Export Web Map](http://desktop.arcgis.com/en/arcmap/latest/tools/server-toolbox/export-web-map.htm) tool is a geoprocessing tool that comes published as a [service](http://resources.arcgis.com/en/help/rest/apiref/gp_exportwebmaptask.html) (`Utilities/PrintingTools`) out of the box with ArcGIS Server. From ESRI's docs:
+> This tool takes the state of a web application (for example, included services, layer visibility settings, and client-side graphics) and returns a printable page layout or basic map of the specified area of interest.
+
+This is most commonly used as a print service for web applications allowing the user to download a PDF of the current map extent.
+
+### The Problem
+AGRC's [base map services](https://gis.utah.gov/data/base-map-and-imagery/) (including the [Google imagery](https://gis.utah.gov/data/google-imagery-license/)) are hosted as WMTS services on `discover.agrc.utah.gov`. In order to use them an organization needs to [register](https://gis.utah.gov/data/base-map-and-imagery/) and request a quad word. To prevent unauthorized use they are each locked down to a specific domain or set of domains. So if someone tries to use your quad word from any other domain or server the request is blocked. This all works well until you send a web map to the PrintingTools service in ArcGIS Server. The web map contains URL's to discover services that have quad words locked down to your web app's domain. When the service requests tiles, it does so from the server that is hosting the service not the browser. This causes discover to reject the request because the origin of the request does not match what the quad word is locked down to.
```

Comment:
any link to within gis.utah.gov should use a relative url and the liquid templating with the base url. See the [readme](https://github.com/agrc/gis.utah.gov#link-to-page) for examples.